crates/bevy_ui/src/ui_node.rs

```diff
@@ -468,6 +468,225 @@ impl Style {
         grid_column: GridPlacement::DEFAULT,
         grid_row: GridPlacement::DEFAULT,
     };
+
+    /// Overlays the `patch` onto `self`, ignoring any fields of the `patch` which are at the `DEFAULT` value.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_ui::{Style, Display, PositionType};
+    /// let patch = Style {
+    ///     display: Display::Grid, // non-default
+    ///     ..Default::default()
+    /// };
+    ///
+    /// let style = Style {
+    ///     position_type: PositionType::Absolute, // non-default
+    ///     ..Default::default()
+    /// };
+    ///
+    /// let patched = style.with(patch);
+    ///
+    /// // `patch`'s default `position_type` does not overwrite `style`'s
+    /// assert_eq!(patched.position_type, PositionType::Absolute);
+    /// assert_eq!(patched.display, Display::Grid);
+    /// ```
+    pub fn with(self, patch: Self) -> Self {
+        Style {
+            display: if patch.display == Self::DEFAULT.display {
+                self.display
+            } else {
+                patch.display
+            },
```

Comment:
For each field, if the `patch` value is the default value, ignore it and keep `self`'s value. Otherwise, use the `patch` value.